client/src/services/AppData/AppDataService.ts

```diff
@@ -1,41 +1,86 @@
-import { MembershipTypes } from "models/Payment"
+import { MembershipTypes } from "models/Payment" // may have to add PricingDetails here
+import fetchClient from "services/OpenApiFetchClient"
+
+export type Prices = {
+  title: string
+  name: MembershipTypes
+  priceString: string
+  originalPrice?: string
+  extraInfo?: string
+}
 
 const AppDataService = {
   getBankPaymentDetails: async function () {
     // TODO: Dynamically fetch and make sure there is appropriate fallback
     const data = { email: "uasc@gmail.com", bankAccount: "unknown" }
     return data
   },
-  getMembershipPricingDetails: async function () {
-    // TODO: Dynamically fetch and make sure there is appropriate fallback
-    const prices: {
-      title: string
-      type: MembershipTypes
-      priceString: string
-      extraInfo?: string
-    }[] = [
+  getMembershipPricingDetails: async function (): Promise<Prices[]> {
+    const { data } = await fetchClient.GET("/payment/membership_prices")
+
+    const fallbackData: Prices[] = [
       {
         title: "UoA Student",
-        type: "uoa_student",
-        priceString: "$75"
+        name: "uoa_student",
+        priceString: "$45",
+        originalPrice: "$65",
+        extraInfo: "Save $20"
       },
       {
         title: "Returning Member",
-        type: "returning_member",
+        name: "returning_member",
         priceString: "$65"
       },
       {
         title: "New Non Student",
-        type: "new_non_student",
+        name: "new_non_student",
         priceString: "$75"
       },
       {
         title: "Non Uoa Student New",
-        type: "non_uoa_student",
+        name: "non_uoa_student",
         priceString: "$95"
       }
     ]
-    return prices
+
+    if (data && data.data) {
+      const transformedData: Prices[] =
+        data.data &&
+        data.data.map((data) => {
+          let displayName
+          switch (data.name) {
+            case "uoa_student":
+              displayName = "UoA Student"
+              break
+            case "non_uoa_student":
+              displayName = "Non-UoA Student"
+              break
+            case "returning_member":
+              displayName = "Returning Member"
+              break
+            case "new_non_student":
+              displayName = "New Non-UoA Student"
+              break
+          }
```

Comment:
You might want to extract this into a util function and use it in `client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx`